morango/sync/syncsession.py

```diff
@@ -423,25 +528,19 @@ def initiate_push(self, sync_filter):
         self._close_transfer_session()
 
     def initiate_pull(self, sync_filter):
-
-        logger.info("Initiating pull sync")
         self._create_transfer_session(False, sync_filter)
 
         if self.current_transfer_session.records_total == 0:
-            logger.info("There are no records to transfer")
             self._close_transfer_session()
             return
 
-        logger.info(
-            "{} records have been queued server side for transfer".format(
-                self.current_transfer_session.records_total
-            )
-        )
-        # pull records and close transfer session upon completion
-        logger.info("Beginning pulling of data...")
-        self._pull_records()
-        logger.info("Completed pull of data")
-        _dequeue_into_store(self.current_transfer_session)
+        with self.pulling.send(
+            transfer_session=self.current_transfer_session
+        ) as status:
+            self._pull_records(status.in_progress.fire)
```

Comment:
Since it's a "named argument" (in the sense that it has a default), may be better to explicitly say `callback=status.in_progress.fire` here.